### PR TITLE
Add citation file.

### PR DESCRIPTION
**Pull request by samcunliffe**
@giordano suggested (and I agree) that it might be a good idea to add a [citation file](https://citation-file-format.github.io/). Assuming future students and users are going to use/cite this code. This makes it a bit easier to get a zenodo (and a DOI) for the software.

There is also the option to request all users of the code cite some document/paper (if such a thing exists).

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,24 @@
+cff-version: 1.2.0
+title: TDMS - Time Domain Maxwell Solver
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
+type: software
+authors:
+  - given-names: Peter
+    family-names: Munro
+    affiliation: Universitu College London
```
Comment (t-young31) [inline CITATION.cff:10]:
```suggestion
    affiliation: University College London
```

File: README.md [hunk 2]
```diff
@@ -121,6 +121,32 @@ $ export OMP_NUM_THREADS=4
 
 Before calling the `tdms` executable.
 
+## Citation
+
+If you used TDMS in your research and found it helpful, please cite this work.
+<!-- [zenodo/FIXME](https://zenodo.org/) -->
```
Comment (samcunliffe) [inline README.md:127]: When we tag version 1, we can put it on Zenodo and get a DOI.

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  - given-names: Tom
+    family-names: Young
+    affiliation: University College London
+    orcid: 'https://orcid.org/0000-0002-8432-7769'
+repository-code: 'https://github.com/UCL/TDMS'
+url: 'https://github-pages.ucl.ac.uk/TDMS'
+keywords:
+  - time-domain
+  - maxwell-equations-solver
+license: GPL-3.0
```
Comment (samcunliffe) [inline CITATION.cff:42]: Can verify that the file is valid with [CFFINT](https://citation-file-format.github.io/cff-initializer-javascript/#/), if you're interested.

**Review by willGraham01 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (samcunliffe) [reply README.md:127]: Yeah, good idea. Let's leave that comment so we remember.

File: README.md [hunk 4]
```diff
@@ -121,6 +121,35 @@ $ export OMP_NUM_THREADS=4
...
+<details>
+<summary>BibTEX</summary>
+
+```bibtex
+@misc{tdms,
+  author       = Munro, Peter and others,
+  title        = {TDMS - The Time-Domain Maxwell Solver},
+  url          = {https://github.com/UCL/TDMS}
+}
+```
```
Comment (giordano) [inline README.md:140]: This is the bibtex entry created by github for the given citation.cff file:
```bibtex
@software{Munro_TDMS_-_Time,
author = {Munro, Peter and Cunliffe, Sam and Giordano, Mosè and Graham, William and Kriezis, Emmanouil and Török, Peter and Young, Tom},
license = {GPL-3.0},
title = {{TDMS - Time Domain Maxwell Solver}},
url = {https://github.com/UCL/TDMS}
}
```
which I think also correctly preserves cases in the title value.

File: README.md [hunk 5]
```diff
@@ -121,6 +121,35 @@ $ export OMP_NUM_THREADS=4
...
+<!-- [contact_address)[mailto:FIXME] -->
+<details>
+<summary>BibTEX</summary>
+
+```bibtex
+@misc{tdms,
+  author       = Munro, Peter and others,
+  title        = {TDMS - The Time-Domain Maxwell Solver},
+  url          = {https://github.com/UCL/TDMS}
+}
```
Comment (samcunliffe) [inline README.md:139]:
```suggestion
@software{tdms,
    author  = {Munro, Peter and others},
    license = {GPL-3.0},
    title   = {{TDMS - Time Domain Maxwell Solver}},
    URL     = {https://github.com/UCL/TDMS}
}
```

File: README.md (hunk 4 above)
Comment (samcunliffe) [reply README.md]: Thanks!  🦅 👀

**Pull request merged**